catalog/core/catalog-core-attributeregistry/src/test/java/ddf/catalog/data/impl/AttributeRegistryImplTest.java

```diff
@@ -13,23 +13,57 @@
  */
 package ddf.catalog.data.impl;
 
-import static org.hamcrest.Matchers.is;
+import static org.hamcrest.core.Is.is;
 import static org.junit.Assert.assertThat;
 
+import java.util.Arrays;
+import java.util.HashSet;
 import java.util.Optional;
+import java.util.Set;
 
 import org.junit.Before;
 import org.junit.Test;
 
 import ddf.catalog.data.AttributeDescriptor;
-import ddf.catalog.data.AttributeRegistry;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.data.impl.types.CoreAttributes;
+import ddf.catalog.data.impl.types.DateTimeAttributes;
+import ddf.catalog.data.types.Core;
 
 public class AttributeRegistryImplTest {
-    private AttributeRegistry registry;
+    private AttributeRegistryImpl registry;
 
+    @SuppressWarnings("unchecked")
     @Before
     public void setup() {
         registry = new AttributeRegistryImpl();
+        registry.registerMetacardType(generateMetacardType());
+    }
+
+    @Test
+    public void testDeregisterMetacardTypeWithDuplicateAttributes() {
```

Comment:
We may also want a test that verifies that deregistering this metacard type will remove its attributes from the registry.